modules/user_accounts/php/edit_user.class.inc

```diff
@@ -817,39 +817,18 @@ class Edit_User extends \NDB_Form
     }
 
     /**
-     * Query database on UserID to check that password hash does not
-     * exist and account is still pending approval.
-     *
      * This ensures that users with account activity cannot be rejected
      * if their approval status is, for some reason, set to pending.
      *
-     * @param string $userID of account to verify
-     *
-     * @return boolean true if 'Password_hash' is null and
-     *         'Pending_approval' is Yes. False, otherwise.
+     * @param \User $user The User to verify
      *
-     * @throws Loris exception if the userID is not found in the database.
```

Comment:
is this no longer important? I know it's a bit of an extra check because what is shown on the front-end comes from the DB

Reply:
It is no longer important. Even more, it was giving this function an additional role/behavior. It should have been called `canRejectAccountAndCheckIfUserExists` otherwise. I don't think that the existence of the user if critical to determine if  a given user can be rejected. If it do not exists, the answer is no.

Reply:
On a side note, if the user do not exists, the User::factory function will return an empty User object with no data because of the select statement. Which is just fine. Unless there is a row in the user_loggin_history with userId = NULL and Successful = 'Y'. We should be fine because the column has a `NOT NULL` constraint.

Reply:
gotchaa, the $user would need to exist in the first place